sdk/eventhub/event-hubs/src/eventData.ts

```diff
@@ -377,10 +377,61 @@ export interface EventData {
   };
 }
 
+/**
+ * Asserts that the provided data conforms to the `EventData` interface.
+ *
+ * This function performs runtime checks on the `data` object to ensure it matches the expected
+ * structure and types defined in the `EventData` interface. If any of the checks fail, it throws
+ * an error with a descriptive message indicating the mismatch.
+ *
+ * @param data - The data object to validate as `EventData`.
+ * @throws \{Error\} Throws an error if the data does not conform to the `EventData` interface.
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+export function assertIsEventData(data: any): asserts data is EventData {
+  if (data.contentType !== undefined && typeof data.contentType !== "string") {
+    throw new Error(
+      `Invalid 'contentType': expected 'string' or 'undefined', but received '${typeof data.contentType}'.`,
+    );
+  }
+
+  if (
+    data.correlationId !== undefined &&
+    typeof data.correlationId !== "string" &&
+    typeof data.correlationId !== "number" &&
+    !Buffer.isBuffer(data.correlationId)
+  ) {
+    throw new Error(
+      `Invalid 'correlationId': expected 'string', 'number', 'Buffer', or 'undefined', but received '${typeof data.correlationId}'.`,
+    );
+  }
+
+  if (
+    data.messageId !== undefined &&
+    typeof data.messageId !== "string" &&
+    typeof data.messageId !== "number" &&
+    !Buffer.isBuffer(data.messageId)
```

Comment:
use `import isBuffer from "is-buffer";`? I am not 100% sure but would assume it works for browser as well. We already using it in EH and SB.